Add quick start guide (#850)
huggingface/huggingface_hub: docs/source/quick-start.mdx

# Quick start

> The Hugging Face Hub is the go-to place for sharing machine learning models, spaces, datasets, and metrics. `huggingface_hub` library helps you interact with the Hub without leaving your development environment. You can create and manage repositories easily and download and upload files from the Hub. 

Reviewer: ```suggestion
The [Hugging Face Hub](http://hf.co/) is the go-to place for sharing machine learning models, demos, datasets, and metrics. `huggingface_hub` library helps you interact with the Hub without leaving your development environment. You can create and manage repositories easily, download and upload files and query for information from the Hub. 
```

> ## Install the Hub library

Reviewer: For consistency with https://huggingface.co/docs/transformers/main/installation I think we could also show how to install with extras for TF and PyTorch as well as installing from source.

Reviewer 2: I don't think that's in the scope of a quick start guide. Also, if we say how to do for TF and PyTorch, then we also give the impression that the hub is only, or mainly for those libraries.

Author: I think if we want to, we can create a separate page with the additional installation options. But it's probably best to keep the quick start guide as lightweight as possible without presenting other options that might distract the user from getting started.

Reviewer: Yes, sounds good!

> To get started, you should install the `huggingface_hub` library. The library provides functionalities that make interacting with the Hub simple. You can install the library with `pip` or `conda`.

Reviewer: I would remove this sentence as this is a bit similar to what is mentioned in the first paragraph.
```suggestion
To get started, you should install the `huggingface_hub` library. You can install the library with `pip` or `conda`.
```

## Create a repository

To create and share files to the Hub, you need to have a Hugging Face account. [Create an account]((https://huggingface.co/join)) if you don't already have one, and then sign in to find your [User Access Token](https://huggingface.co/docs/hub/security#user-access-tokens) in your Settings. The User Access Token is used to authenticate your identity to the Hub.

<Tip>

You can also provide your token to our functions and methods. This way you don't need to store your token anywhere.

</Tip>

1. Log in to your Hugging Face account with the following command:

```bash
huggingface-cli login
```

Or if you prefer to work from a Jupyter or Colaboratory notebook, then login with [`notebook_login`]:

```py
>>> from huggingface_hub import notebook_login
>>> notebook_login()
```

2. Enter your User Access Token to authenticate your identity to the Hub.

After you've created an account, create a repository with the [`create_repo`] function:

```py
>>> from huggingface_hub import create_repo
>>> create_repo(repo_name="super-cool-model")
```

If you want your repository to be private, then:

```py
>>> from huggingface_hub import create_repo

> >>> create_repo(repo_name="super-cool-model", private=True)

Reviewer: On these examples consider showing `repo_id`. `name` is deprecated since https://github.com/huggingface/huggingface_hub/pull/733. That way we're also consistent across all methods.

# Quick start

> The Hugging Face Hub is the go-to place for sharing machine learning models, datasets, and metrics. To help you get the most out of the Hub, we created the `huggingface_hub` library so you can interact with the Hub without leaving your development environment. You can create and manage repositories easily and download and upload files from the Hub. 

Reviewer 2: ```suggestion
The Hugging Face Hub is the go-to place for sharing machine learning models, spaces, datasets, and metrics. `huggingface_hub` library helps you interact with the Hub without leaving your development environment. You can create and manage repositories easily and download and upload files from the Hub. 
```

## Install the Hub library

> To get started, you should install the `huggingface_hub` library. The library provides functions that make interacting with the Hub simple. Run the following command to install the library:

Reviewer 2: ```suggestion
To get started, you should install the `huggingface_hub` library. The library provides functionalities that make interacting with the Hub simple. You can install the library using `pip` or `conda`:
```

> ```bash
> >>> pip install huggingface_hub
> ```

Reviewer 2: ````suggestion
Using `pip`:

```bash
pip install huggingface_hub
```

Or using `conda`:

``` bash
conda install -c conda-forge huggingface_hub
```
````

## Download files from the Hub

> Repositories store versioned model and dataset files anyone can download. You can use the [`hf_hub_url`] function to download files. This function will download and cache a file on your local disk. The next time you need that file, it will load from your cache, so you don't need to re-download it.

Reviewer 2: ```suggestion
Repositories on the hub are git version controlled and users can download a single file, or the whole repository. You can use the [`hf_hub_download`] function to download files. This function will download and cache a file on your local disk. The next time you need that file, it will load from your cache, so you don't need to re-download it.
```

You will need the repository id and the filename of the file you want to download. For example, to download the [Pegasus](https://huggingface.co/google/pegasus-xsum) model configuration file: 

```py
>>> from huggingface_hub import hf_hub_download
>>> hf_hub_download(repo_id="google/pegasus-xsum", filename="config.json")
```

To download a specific version of the file, use the `revision` parameter to specify the branch name, tag, or commit hash. If you choose to use the commit hash, it must be the full-length hash instead of the shorter 7-character commit hash: 

```py
>>> from huggingface_hub import hf_hub_download

> >>> hf_hub_download(repo_id="google/pegasus-xsum", filename="config.json", revision="4d33b01d79672f27f001f6abade33f22d993b151")

Reviewer 2: ```suggestion
>>> hf_hub_download(
....     repo_id="google/pegasus-xsum",
....     filename="config.json",
....     revision="4d33b01d79672f27f001f6abade33f22d993b151"
.... )
```

```

Finally, you can choose where to cache a file with the `cache_dir` parameter:

```py
>>> from huggingface_hub import hf_hub_download
>>> hf_hub_download(repo_id="google/pegasus-xsum", filename="config.json", cache_dir="/my/cool/cache")

> ```

Reviewer 2: I don't think we need this, there are many options and users should refer to the API docs for further details. You can mention that they can check the docs for more options.

## Create a repository

To create and share files to the Hub, you need to have a Hugging Face account. [Create an account]((https://huggingface.co/join)) if you don't already have one, and then sign in to find your [User Access Token](https://huggingface.co/docs/hub/security#user-access-tokens) in your Settings. The User Access Token is used to authenticate your identity to the Hub.

1. Log in to your Hugging Face account with the following command:

```bash
huggingface-cli login
```

Or if you prefer to work from a Jupyter or Colaboratory notebook, then login with [`notebook_login`]:

```py
>>> from huggingface_hub import notebook_login
>>> notebook_login()

> ```

Reviewer 2: Users can also chose to provide the token to methods/functions, they don't need to have it stored anywhere. Alternatively, they can use our methods to store the token.

2. Enter your User Access Token to authenticate your identity to the Hub.

After you've created an account, create a repository with the [`create_repo`] function:

```py
>>> from huggingface_hub import create_repo
>>> create_repo(repo_name="super-cool-model")

> ```

Reviewer 2: I rather tell users to use `HfApi().create_repo` rather than the provided method.

Author: So do this instead (and the same for below)?

```py
from huggingface_hub import HfApi

api = HfApi()
api.create_repo(repo_id="super-cool-model")
```

Reviewer: I think there was a strong push from folks to use `create_repo` instead of `HfApi()` due to confusions some time ago.

Reviewer 2: `create_repo` _is_ `HfApi()`, it's just a shorthand for the default instance of `HfApi()`. The API would look much nicer once we store the token in the instance rather than somewhere on disk.

If you want your repository to be private, then:

```py
>>> from huggingface_hub import create_repo
>>> create_repo(repo_name="super-cool-model", private=True)
```

Private repositories will not be visible to anyone except yourself.

## Share files to the Hub

Use the [`upload_file`] function to add a file to your newly created repository. You need to specify:

1. The path of the file to upload.
2. The path of the file in the repository.
3. The repository id of where you want to add the file.

```py
>>> from huggingface_hub import upload_file
>>> upload_file(path_or_fileobj="/home/lysandre/dummy-test/README.md", path_in_repo="README.md", repo_id="lysandre/test-model")

> ```

Reviewer 2: same here, use `HfApi().upload_file()`

```py
>>> from huggingface_hub import hf_hub_download
>>> hf_hub_download(
...     repo_id="google/pegasus-xsum", 
...     filename="config.json", 
...     revision="4d33b01d79672f27f001f6abade33f22d993b151"
... )
```

For more details and options, see the API reference for [`hf_hub_download`].

## Create a repository

> To create and share files to the Hub, you need to have a Hugging Face account. [Create an account]((https://huggingface.co/join)) if you don't already have one, and then sign in to find your [User Access Token](https://huggingface.co/docs/hub/security#user-access-tokens) in your Settings. The User Access Token is used to authenticate your identity to the Hub.

Reviewer 3: [Create an account]((https://huggingface.co/join)) there are two parenthesis which redirects to (https://huggingface.co/join)
